src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,130 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class TaskList {
+    protected ArrayList<Task> list;
+
+    /**
+     * Construct a TaskList object
+     */
+    public TaskList() {
+        list = new ArrayList<>();
+    }
+
+    /**
+     * Construct a TaskList object
+     * @param list an ArrayList of Task
+     */
+    public TaskList(ArrayList<Task> list) {
+        this.list = list;
+    }
+
+    /**
+     * Construction a TaskList object
+     * @param reader a BufferedReader to read the ./data/main.java.duke.txt
+     * @throws IOException
+     */
+    public TaskList(BufferedReader reader) throws IOException {
+        if (reader == null) {
+            list = new ArrayList<>();
+        } else {
+            list = readTextFile2List(reader);
+        }
+    }
+
+    /**
+     * Reads all the Tasks in ./data/main.java.duke.txt and puts all the Tasks in an ArrayList
+     * @param reader a BufferedReader for reading ./data/main.java.duke.txt
+     * @return an ArrayList of all Tasks in ./data/main.java.duke.txt
+     * @throws IOException
+     */
+    public static ArrayList<Task> readTextFile2List(BufferedReader reader) throws IOException {
```

Comment:
Any reason why you abbreviate the method? readTextFile2List() struck me as a method to read a second text file into the list before I realized the short-form.